### PR TITLE
fix(ssr): do not render class attribute when dynamic value is null, empty or undefined

### PR DESCRIPTION
**Pull request by cardoso**
## Details

Fixes two expected failures:

`'attribute-class/unstyled/dynamic/index.js'`
`'attribute-class/with-scoped-styles-only-in-parent/dynamic/index.js'`

## Does this pull request introduce a breaking change?

-   😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

-   🔬 Yes, it does include an observable change.

The class attribute will not be rendered when the dynamic value is null, empty or undefined

### CODE REVIEW

File: packages/@lwc/ssr-compiler/src/__tests__/fixtures.spec.ts [hunk 1]
```diff
@@ -11,10 +11,9 @@ import { rollup } from 'rollup';
 import lwcRollupPlugin from '@lwc/rollup-plugin';
 import { testFixtureDir, formatHTML } from '@lwc/test-utils-lwc-internals';
 import { serverSideRenderComponent } from '@lwc/ssr-runtime';
-import { DEFAULT_SSR_MODE } from '@lwc/shared';
+import { DEFAULT_SSR_MODE, type CompilationMode } from '@lwc/shared';
 import { expectedFailures } from './utils/expected-failures';
 import type { FeatureFlagName } from '@lwc/features/dist/types';
-import type { CompilationMode } from '../index';
```
Comment (cardoso) [inline packages/@lwc/ssr-compiler/src/__tests__/fixtures.spec.ts:17]: Missed this in #4933

File: packages/@lwc/ssr-runtime/src/render.ts [hunk 2]
```diff
@@ -37,9 +37,17 @@ function renderAttrsPrivate(
...
+        if (attrName === 'class') {
+            if (attrValue === '') {
+                // If the class attribute is empty, we don't render it.
+                continue;
+            }
+
+            if (combinedScopeToken) {
+                attrValue += ' ' + combinedScopeToken;
+                hasClassAttribute = true;
+            }
```
Comment (cardoso) [inline packages/@lwc/ssr-runtime/src/render.ts:50]: I don't normally like to use `continue`, but some refactoring would be needed to avoid it here.

**Review by nolanlawson — APPROVED** [review]
Great find! Thanks! 🧸

**Pull request merged**